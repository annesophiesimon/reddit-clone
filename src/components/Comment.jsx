const Comment = (comment) => {
  const { name, time, content, isHidden } = comment;
  return (
    <div className={`bg-[#f2f2f2] shadow-lg p-3 m-5 rounded-lg ${isHidden ? 'hidden' : ''}`}>
      <div>
        <div className="flex justify-between">
          <h1 className="font-bold text-[#212427] text-sm"> {name} </h1>
          <p className="text-slate-500 text-xs"> {time}</p>
        </div>
        <p className=" text-[#212427]">{content}</p>
      </div>
    </div>
  );
};

export default Comment;
