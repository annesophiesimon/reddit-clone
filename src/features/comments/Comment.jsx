import { convertTime } from '../../helpers';
/* eslint-disable react/prop-types */
// add nested comments ?
const Comment = (props) => {
  const { comments } = props;
  return (
    <>
      {comments?.length > 0 &&
        comments.map((comment, index) => (
          <div key={index} className="bg-[#f2f2f2] shadow-lg p-3 m-5 rounded-lg">
            <div>
              <div className="flex justify-between">
                <h1 className="font-bold text-[#212427] text-sm"> {comment.author}</h1>
                <p className="text-slate-500 text-xs"> {convertTime(comment.created_utc)}</p>
              </div>
              <p className=" text-[#212427]"> {comment.body}</p>
            </div>
          </div>
        ))}
    </>
  );
};

export default Comment;
