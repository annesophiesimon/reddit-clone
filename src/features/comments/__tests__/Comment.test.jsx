import Comment from '../Comment';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

let date = new Date(Date.now());
date.setMinutes(date.getMinutes() - 30);
date = Math.floor(date / 1000);
const comments = [
  {
    id: 1,
    author: 'Joel',
    created_utc: date,
    body: 'I like this pattern'
  }
];
describe('Comment', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Comment comments={comments} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders empty elements when comments is empty', () => {
    const { container } = render(<Comment comments={[]} />);
    expect(container).toBeEmptyDOMElement();
  });

  it('renders components with author,time and body', () => {
    render(<Comment comments={comments} />);
    const nameElement = screen.getByText('Joel');
    const timeElement = screen.getByText('30 minutes ago');
    const contentElement = screen.getByText('I like this pattern');

    expect(nameElement).toBeInTheDocument();
    expect(timeElement).toBeInTheDocument();
    expect(contentElement).toBeInTheDocument();
  });
});
