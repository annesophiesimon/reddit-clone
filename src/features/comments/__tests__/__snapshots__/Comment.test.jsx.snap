// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment renders correctly 1`] = `
<div
  className="bg-[#f2f2f2] shadow-lg p-3 m-5 rounded-lg"
>
  <div>
    <div
      className="flex justify-between"
    >
      <h1
        className="font-bold text-[#212427] text-sm"
      >
         
        Joel
      </h1>
      <p
        className="text-slate-500 text-xs"
      >
         
        30 minutes ago
      </p>
    </div>
    <p
      className=" text-[#212427]"
    >
       
      I like this pattern
    </p>
  </div>
</div>
`;
